automation/src/main/java/org/wso2/testgrid/automation/Test.java

```diff
@@ -142,18 +145,24 @@ public TestScenario execute(String testLocation, DeploymentCreationResult input)
         testExecutor.init(testLocation, getTestName(), testScenario);
 
         if (!StringUtil.isStringNullOrEmpty(preScenarioScript)) {
-            String result = testExecutor.executeEnvironmentScript(Paths.get(preScenarioScript), input);
-            if (result.contains("OK")) {
+            int result = testExecutor.executeEnvironmentScript(Paths.get(preScenarioScript), input);
+            if (result == 0) {
                 testScenario.setIsPreScriptSuccessful(true);
             }
         }
+        try {
+            for (String script : scripts) {
+                testExecutor.execute(script, input);
+            }
+        } catch (Exception e) {
```

Comment:
Normally, we avoid catching the generic exceptions. It seems like TestAutomationException has to be handled within a try-catch block.

Reply:
A generic exception can arise from too many sources. If you catch a generic exception in a context that does not allow any exceptions.

Reply:
In TestGrid we need to continue running scenarios regardless of the exception thrown. So we have catch all exceptions and continue the flow.